### PR TITLE
Disputer should correctly simulate reward withdrawals

### PR DESCRIPTION
**Pull request by mrice32**
Disputer was failing to detect when a withdraw will fail because of an existing [bug](https://github.com/ethereum/solidity/issues/4840) in `solidity`/`web3`.

### CODE REVIEW

File: disputer/disputer.js [hunk 1]
```diff
@@ -197,6 +197,13 @@ class Disputer {
       let withdrawAmount;
       try {
         withdrawAmount = await withdraw.call({ from: this.account });
+        // See: https://github.com/ethereum/solidity/issues/4840 for the reason this fix is necessary.
+        if (
+          withdrawAmount.rawValue.toString() ===
```
Comment (nicholaspai) [inline disputer/disputer.js:202]: Look at `common/ContractUtils` for a `revertWrapper` module you can use that checks for this hard-coded value. We use it in `reporters/GlobalSummaryReporter` to check if a price has resolved for example.

Comment (chrismaree) [reply disputer/disputer.js:202]: ye, I agree would be ideal to not have this hardcoded here like this if it's something we deal with in other contexts.

Comment (mrice32) [reply disputer/disputer.js:202]: Yeah, the initial plan was to just get it merged as fast as possible to patch the live bots. But now that we've done the hot fix, I'll move this logic elsewhere.

Comment (mrice32) [reply disputer/disputer.js]: Done.

File: disputer/test/Disputer.js [hunk 2]
```diff
@@ -280,6 +280,13 @@ contract("Disputer.js", function(accounts) {
     await disputer.queryAndDispute();
     assert.equal(spy.callCount, 2); // Two info level events for the two disputes.
 
+    // Before the dispute is resolved, the bot should simulate the withdrawal, determine that it will fail, and
+    // continue to wait.
+    await disputer.queryAndWithdrawRewards();
```
Comment (nicholaspai) [inline disputer/test/Disputer.js:285]: Nice lean way to add in this unit test

**Comment by mrice32** [comment]
> Great catch! We should re-use the `common/ContractUtils:revertWrapper()` method to handle functions that will fail.
> 
> Additionally, I was surprised to find that we don't have a unit test in `disputers/test` that tests when `queryAndWithdrawRewards` is called while there is a pending, but not withdraweable, dispute. Can you add one that just tests if the the `disputer` doesn't attempt to withdraw from disputes that will fail and fails silently?

Note: agree that it's weird we didn't have a test for this. In building the test, I came to the conclusion that I don't think this can be replicated in tests because this seems to be a node-specific issue. On ganache, the `.call` seems to be reverting as expected. Put another way, the test passes with the old implementation. I added logs to be sure that that initial catch was actually being hit and it was (to rule out an issue in the implementation of my test).

**Pull request merged**